tasks/new_e2e_tests.py

```diff
@@ -321,81 +324,86 @@ def post_process_output(path: str, test_depth: int = 1):
     TestPackages/run_centos
     TestPackages/run_debian
     We should set test_depth to 2 to avoid mixing all the logs of the different tested platform
+
+    Returns:
+        A list of (package name, test name, logs) tuples
     """
 
     def is_parent(parent: list[str], child: list[str]) -> bool:
+        if len(parent) > len(child):
+            return False
+
         for i in range(len(parent)):
             if parent[i] != child[i]:
                 return False
+
         return True
 
-    logs_per_test = {}
     with open(path) as f:
-        all_lines = f.readlines()
+        lines = [json.loads(line) for line in f]
 
-        # Initalize logs_per_test with all test names
-        for line in all_lines:
-            json_line = json.loads(line)
-            if "Package" not in json_line or "Test" not in json_line or "Output" not in json_line:
-                continue
-            splitted_test = json_line["Test"].split("/")
-            if len(splitted_test) < test_depth:
-                continue
-            if json_line["Package"] not in logs_per_test:
-                logs_per_test[json_line["Package"]] = {}
+    lines = [
+        json_line for json_line in lines if "Package" in json_line and "Test" in json_line and "Output" in json_line
```

Comment:
```suggestion
        json_line for json_line in lines if all(word in json_line for words in ["Package", "Test", "Output"])
```
I'm not 100% convinced this is better, still I do the suggestion, up to you to decide 😄

Reply:
I'm not sure too as it adds some complexity, I prefer the initial one